etna/transforms/math/differencing.py

```diff
@@ -194,12 +199,9 @@ def _reconstruct_test(self, df: pd.DataFrame, columns_to_inverse: Set[str]) -> p
         result_df = df.copy()
 
         # check that test is right after the train
-        expected_min_test_timestamp = pd.date_range(
-            start=self._test_init_df.index.max(),  # type: ignore
-            periods=2,
-            freq=pd.infer_freq(self._train_timestamp),
-            closed="right",
-        )[0]
+        expected_min_test_timestamp = timestamp_range(
```

Comment:
Doesn't look the same as previous version

Reply:
The difference:
- Add using `self._freq` saved during `fit`
- Remove `closed="right"` and use `[-1]`
- Use `self._test_init_df.index[-1]` instead of `self._test_init_df.index.max()`

Reply:
Don't get why we use [-1] instead of [0]